tl.tl

```diff
@@ -259,6 +259,11 @@ function tl.lex(input: string): {Token}
          end
       end
    end
+
+   if #tokens > 0 and tokens[#tokens].tk == nil then
+      drop_token()
+   end
```

Comment:
This here is so that we would not add an invalid (unfinished) last token, in case of a lexer error. Because then the parser would crash (because of missing `.tk` field).

Although it would be better to raise a lexer error here instead (as there shouldn't be unfinished tokens).

Reply:
> Although it would be better to raise a lexer error here instead (as there shouldn't be unfinished tokens). Because now a program with some unfinished token (like an unclosed long string/comment) would be treated by tl as a valid tl program (and tl would only process the part of the program up until that unfinished token).

Agreed, it would be better to make it return nil+error and handle it in the proper places.